xchange-core/src/main/java/org/knowm/xchange/currency/Currency.java

```diff
@@ -442,7 +442,7 @@ public String getDisplayName() {
   @Override
   public String toString() {
 
-    return code;
+    return attributes.commonCode;
```

Comment:
currencies created on the fly don't have a common code

Reply:
fixed.